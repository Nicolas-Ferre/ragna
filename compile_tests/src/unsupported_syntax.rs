fn main() {}

#[ragna::gpu]
mod gpu {
    enum Enum {}

    fn run() {
        static LOCAL_GLOB: i32 = 0;
        let mut var: i32;
        var = 0;
        let _ = 0;
        loop {}
    }

    fn invalid_fn(_param: i32) -> i32 {
        unimplemented!()
    }
}
