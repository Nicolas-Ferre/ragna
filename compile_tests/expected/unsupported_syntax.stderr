error: unsupported item
 --> compile_tests/src/unsupported_syntax.rs:5:5
  |
5 |     enum Enum {}
  |     ^^^^

error: unsupported item
 --> compile_tests/src/unsupported_syntax.rs:8:9
  |
8 |         static LOCAL_GLOB: i32 = 0;
  |         ^^^^^^

error: unsupported variable definition syntax
  --> compile_tests/src/unsupported_syntax.rs:11:13
   |
11 |         let _ = 0;
   |             ^

error: unsupported expression
  --> compile_tests/src/unsupported_syntax.rs:12:9
   |
12 |         loop {}
   |         ^^^^

error: unsupported function params
  --> compile_tests/src/unsupported_syntax.rs:15:19
   |
15 |     fn invalid_fn(_param: i32) -> i32 {
   |                   ^^^^^^

error: unsupported function output
  --> compile_tests/src/unsupported_syntax.rs:15:35
   |
15 |     fn invalid_fn(_param: i32) -> i32 {
   |                                   ^^^

error: unsupported expression
  --> compile_tests/src/unsupported_syntax.rs:16:9
   |
16 |         unimplemented!()
   |         ^^^^^^^^^^^^^

error[E0381]: used binding `var` isn't initialized
  --> compile_tests/src/unsupported_syntax.rs:10:9
   |
9  |         let mut var: i32;
   |             ------- binding declared here but left uninitialized
10 |         var = 0;
   |         ^^^ `var` used here but it isn't initialized
   |
help: consider assigning a value
   |
9  |         let mut var: i32 = /* value */;
   |                          +++++++++++++
